### ISSUE TITLE
Add more words to awesome/spell-check

### ISSUE THREAD

**Issue by transitive-bullshit** [sindresorhus/awesome-lint#36]
I'm going to track the master list of words we've officially decided to spell check in this first comment. As we agree on additional words to add to the spell check list, I'll update this list accordingly.

Done:
- [x] `Node.js`
- [x] `Stack Overflow`
- [x] `JavaScript`
- [x] `macOS`
- [x] `YouTube`
- [x] `GitHub`

Todo:
- [ ] `FFmpeg`
- [ ] `Open Source`

**Comment by sindresorhus** [comment]
And `FFmpeg`

**Comment by itaisteinherz** [comment]
Regarding #40 and `FFmpeg`, note that `awesome-lint` currently lints just text nodes (more info about this [here](https://github.com/syntax-tree/unist-util-visit-parents#visittree-test-visitor-reverse) and [here](https://github.com/syntax-tree/unist-util-is)):
https://github.com/sindresorhus/awesome-lint/blob/3b4f6f3cd2870fb9f5d42ff42045b69db8d76a44/rules/spell-check.js#L10-L11
Which means that even if this is the markdown file:
```markdown
This is a document about `ffmpeg`.
```
`ffmpeg` wouldn't be linted. To solve this we could add more regexes which match `ffmpeg` (case sensitive) as a text node and `ffmpeg` (case insensitive) as a code node and assert that the match is `ffmpeg` as a code node (this requires visiting every code node in the AST, in addition to the text nodes).

**Comment by transitive-bullshit** [comment]
I really appreciate you helping out with this @itaisteinherz.

Your logic makes sense. I just wasn't sure without manually testing whether inline code snippets contained text nodes as children in which case `ffmpeg` couldn't necessarily be differentiated from a text node "ffmpeg".

With that being said, given that you've added the relevant unit test to cover this case, I think my reservations are handled, though I'll let @sindresorhus make the final call on this scenario.

**Comment by itaisteinherz** [comment]
Some more ideas for terms we could spell-check:
 - `npm`
 - `GitLab`
 - `AVA` (not sure what's the preferred case here - `AVA` or `ava`)
 - `XO` (same comment as above)
 - `Linux` (since we've already added one OS; we probably shouldn't add `Windows` for now, as it's a widely used word in contexts other than the OS, and it would complicate things)
 - `iOS`, `watchOS`, `tvOS` (to accompany `macOS`)
 - `Docker`
 - DB names: `MySQL`, `MongoDB`, `Neo4j`, etc.
 - Popular file formats - `PNG`, `PDF`, etc.

We could also add some more language names (i.e. `Swift`, `Python`, etc.), since we already have `Node.js`.

There are a lot of words and terms we could spell-check, and so I think there should be some guidelines as to which words we spell-check and which words we don't.

**Comment by transitive-bullshit** [comment]
> There are a lot of words and terms we could spell-check, and so I think we should decide on some guidelines as to which words we spell-check and which words we don't.

Agreed. imho I'd rather keep the list very small and targeted for now until we figure out what to do with #39 since I really want to be careful about the possibility of adding false positives before an initial integration with the greater awesome list community.

**Comment by sindresorhus** [comment]
@itaisteinherz All your suggestions look good to me. For AVA and XO, it's all uppercase.
